src/core/lib/event_engine/windows/windows_listener.cc

```diff
@@ -0,0 +1,323 @@
+                       sizeof(tmp_listener_socket));
+  if (err != 0) {
+    gpr_log(GPR_ERROR, "%s",
+            GRPC_WSA_ERROR(WSAGetLastError(), "setsockopt").ToString().c_str());
+    return close_socket_and_restart();
+  }
+  EventEngine::ResolvedAddress peer_address;
+  int peer_name_len = EventEngine::ResolvedAddress::MAX_SIZE_BYTES;
+  err =
+      getpeername(accept_socket_, const_cast<sockaddr*>(peer_address.address()),
+                  &peer_name_len);
+  if (err != 0) {
+    gpr_log(
+        GPR_ERROR, "%s",
+        GRPC_WSA_ERROR(WSAGetLastError(), "getpeername").ToString().c_str());
+    return close_socket_and_restart();
+  }
+  peer_address =
+      EventEngine::ResolvedAddress(peer_address.address(), peer_name_len);
+  auto addr_uri = ResolvedAddressToURI(peer_address);
+  std::string peer_name = "unknown";
+  if (!addr_uri.ok()) {
+    // TODO(hork): test an early exit/restart here with end2end tests
+    gpr_log(GPR_ERROR, "invalid peer name: %s",
+            addr_uri.status().ToString().c_str());
+  } else {
+    peer_name = *addr_uri;
+  }
+  auto endpoint = std::make_unique<WindowsEndpoint>(
+      peer_address, listener_->iocp_->Watch(accept_socket_),
+      listener_->memory_allocator_factory_->CreateMemoryAllocator(
+          absl::StrFormat("listener endpoint %s", peer_name)),
+      listener_->config_, listener_->executor_);
+  listener_->accept_cb_(
+      std::move(endpoint),
+      listener_->memory_allocator_factory_->CreateMemoryAllocator(
+          absl::StrFormat("listener accept cb for %s", peer_name)));
+  close_socket_and_restart(/*do_close_socket=*/false);
+}
+
+WindowsEventEngineListener::SinglePortSocketListener::SinglePortSocketListener(
+    WindowsEventEngineListener* listener, LPFN_ACCEPTEX AcceptEx,
+    grpc_core::OrphanablePtr<WinSocket> win_socket, int port,
+    EventEngine::ResolvedAddress hostbyname)
+    : AcceptEx(AcceptEx),
+      listener_(listener),
+      on_accept_([this]() { OnAcceptCallbackImpl(); }),
+      port_(port),
+      listener_socket_(std::move(win_socket)),
+      listener_sockname_(hostbyname) {}
+
+absl::StatusOr<WindowsEventEngineListener::SinglePortSocketListener::
+                   PrepareListenerSocketResult>
+WindowsEventEngineListener::SinglePortSocketListener::PrepareListenerSocket(
+    SOCKET sock, const EventEngine::ResolvedAddress& addr) {
+  auto fail = [&](absl::Status error) -> absl::Status {
+    GPR_ASSERT(!error.ok());
+    auto addr_uri = ResolvedAddressToURI(addr);
+    error = grpc_error_set_int(
+        grpc_error_set_str(
+            GRPC_ERROR_CREATE_REFERENCING("Failed to prepare server socket",
+                                          &error, 1),
+            grpc_core::StatusStrProperty::kTargetAddress,
+            addr_uri.ok() ? *addr_uri : addr_uri.status().ToString()),
+        grpc_core::StatusIntProperty::kFd, static_cast<intptr_t>(sock));
+    if (sock != INVALID_SOCKET) closesocket(sock);
+    return error;
+  };
+
```

Comment:
channeling my Mark: replace the blank space with a comment, or remove it altogether

Reply:
blank removed :-). Hopefully the next line is sufficiently self-documenting.